kamatera/disk_test.go

```diff
@@ -0,0 +1,57 @@
+package kamatera
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_calDiskChangeOperation(t *testing.T) {
+	tests := []struct {
+		name        string
+		o           []interface{}
+		n           []interface{}
+		expected    diskOperation
+		expectedErr error
+	}{
+		{
+			name:     "update only",
+			o:        []interface{}{123.0, 456.0},
+			n:        []interface{}{123.0, 457.0},
+			expected: diskOperation{update: map[int]float64{1: 457}},
+		},
+		{
+			name:     "remove only",
+			o:        []interface{}{123.0, 456.0},
+			n:        []interface{}{123.0},
+			expected: diskOperation{remove: []int{1}},
+		},
+		{
+			name: "add only",
+			o: []interface{}{123.0},
+			n: []interface{}{123.0, 456.0},
+			expected: diskOperation{add: []float64{456}},
+		},
+		{
+			name: "update and add",
+			o: []interface{}{123.0},
+			n: []interface{}{124.0, 456.0},
+			expected: diskOperation{add: []float64{456}, update: map[int]float64{0: 124}},
+		},
+		{
+			name: "update and remove",
+			o: []interface{}{123.0, 456.0},
+			n: []interface{}{124.0},
+			expected: diskOperation{remove: []int{1}, update: map[int]float64{0: 124}},
+		},
```

Comment:
please add some more test cases for edge-cases like:
* both disks having the same size
* 3 / 4 disks
* error cases

I'm especially worried about cases where we might accidentially delete customer data.. maybe it's worth to also check how it is handled in other terraform drivers

Reply:
Good catch! I will do it now.

Reply:
Some new cases are added. Please check them.